src/commons/util/mod.rs

```diff
@@ -36,6 +35,11 @@ impl KrillVersion {
         Self::from_str(KRILL_VERSION).unwrap()
     }
 
+    /// Make a notation friendly to namespaces for upgrades.
+    pub fn hyphen_notated(&self) -> String {
+        format!("{}-{}-{}{}", self.major, self.minor, self.patch, self.release_type)
```

Comment:
There's in principle nothing that guarantees that `self.release_type` will be in a format friendly to namespaces, but I guess we are not going to ever put something silly in there, and if we do we will probably notice it anyway.

Reply:
In principle, these values come from the `version` entry in Cargo.toml. And we use a sane scheme for the release_type postfix.